packages/suite/src/hooks/suite/usePinModal.ts

```diff
@@ -1,28 +1,47 @@
+import { useState } from 'react';
+
 import { ButtonRequest } from '@suite-common/suite-types';
 import { selectDeviceButtonRequests } from '@suite-common/wallet-core';
+import TrezorConnect, { UI } from '@trezor/connect';
 
-import { useSelector } from 'src/hooks/suite';
+import { onPinSubmit } from 'src/actions/suite/modalActions';
+import { useDispatch, useSelector } from 'src/hooks/suite';
 
 const NEW_PIN_REQUEST_TYPES = ['PinMatrixRequestType_NewFirst', 'PinMatrixRequestType_NewSecond'];
 const NEW_WIPE_CODE_REQUEST_TYPES = [
     'PinMatrixRequestType_WipeCodeFirst',
     'PinMatrixRequestType_WipeCodeSecond',
 ];
 
-export const usePinWithoutSelector = (buttonRequests: Pick<ButtonRequest, 'code'>[]) => {
-    const pinRequestType = buttonRequests[buttonRequests.length - 1];
-    const invalidCounter = buttonRequests.filter(r => r.code === 'ui-invalid_pin').length || 0;
+const usePinWithoutSelector = (buttonRequests: Pick<ButtonRequest, 'code'>[]) => {
+    const dispatch = useDispatch();
 
-    const isWipeCode =
+    const [pin, setPin] = useState('');
+
+    const pinRequestType = buttonRequests[buttonRequests.length - 1];
+    const isSettingNewWipeCode =
         pinRequestType?.code && NEW_WIPE_CODE_REQUEST_TYPES.includes(pinRequestType?.code);
-    const isRequestingNewPinCode =
+    const isSettingNewPin =
         pinRequestType?.code && NEW_PIN_REQUEST_TYPES.includes(pinRequestType.code);
-    const isPinInvalid = invalidCounter > 0;
+
+    const onCancel = () =>
+        isSettingNewWipeCode
+            ? TrezorConnect.cancel('wipe-cancelled')
+            : TrezorConnect.cancel('pin-cancelled');
+
+    const handlePinSubmit = () => {
+        dispatch(onPinSubmit(pin));
+        setPin('');
+    };
 
     return {
-        isWipeCode,
-        isRequestingNewPinCode,
-        isPinInvalid,
+        isSettingNewWipeCode,
+        isSettingNewPin,
+        hasInvalidAttempts: buttonRequests.filter(r => r.code === UI.INVALID_PIN).length > 0,
```

Comment:
Tiny nit for less code:
```suggestion
        hasInvalidAttempts: buttonRequests.some(r => r.code === UI.INVALID_PIN),
```